src/librustc/mir/interpret/error.rs

```diff
@@ -137,6 +137,7 @@ impl<'tcx> ConstEvalErr<'tcx> {
     ) -> Result<DiagnosticBuilder<'tcx>, ErrorHandled> {
         let must_error = match self.error {
             err_inval!(Layout(LayoutError::Unknown(_))) |
+            err_inval!(Layout(LayoutError::Unsized(_))) |
```

Comment:
An unsized type isnt generic, so returning TooGeneric doesnt make sense.

Reply:
But that's also why this entire approach looks odd... a generic type will eventually be monomorphized, so that's a program that cannot be executed now but can be executed later. But if the type is unsized now it will always be unsized, right, so why does it make any sense to delay the error?

Reply:
yea that should just be reported via the machinery below, which should simplify the rest